### ISSUE TITLE
image recognition with CNN

### ISSUE THREAD

**Issue by yuchenhou** [yuchenhou/elephant#6]
### Scenario: recognize an image
Given an image (a vector)
Predict the object in the image (a vector)

### Implementation: CNN (convolutional neural net)
The CNN reads the whole image at once, and predict the object in the image. The object is represented by a 1000D vector.

### Highlights:
 0. small and replicated filters for each slice: reduce complexity and prevent overfitting
 0. straightforward vector representation of images (with scaling and cropping)

### Question:
 0. what's the exact connection between 2 convolutional layers?
 0. what's the best way to represent the object:
  0. the object classes are mutual exclusive: e.g., if cat = (1, 0, 0, 0...), tiger = (0, 1, 0, 0,...), the object can only be one of cat and tiger.
  0. the object classes are independent: a multi-task prediction with many independent labels like cat, tiger, table, ball, etc.
  0. the object classes are correlated: e.g., if cat = (3, 0, 11, 0...), tiger = (3, 1, 0, 0,...), dog = (4, 0, 11, 0...)... every tiger is a cat, dog is closer to cat than to tiger? this one should be the best.

**Comment by lbholder** [comment]
My understanding of convolution networks is that the input nodes essentially perform a convolution around the pixel corresponding to that input node. The convolution usually computes some value(s) related to the neighborhood of that pixel, e.g., change in intensity from left neighbor to right (i.e., an edge in the image).

For graphs, assuming we have an input node for each node in the graph (a big assumption, and perhaps impossible for massive graphs), such a graph-convolution node would compute some value based on the node's attributes and neighboring nodes. For massive graphs, perhaps a sampling of nodes is sufficient. So, if we have a million-node input deep network, and our graph has a billion nodes, then we sample a million from the billion.

**Comment by yuchenhou** [comment]
Looks like the "convolution" refers to the convolution of the filters to the input.
